### ISSUE TITLE
ServiceAccount not getting annotations from global annotations

### ISSUE THREAD

**Issue by mk2134226** [pingidentity/helm-charts#287]
If i use  this

```
global:
  annotations:
      eks.amazonaws.com/role-arn: arn:aws:iam::xxxx:role/xxxxxxx-pingelm-pap-irsa

```

It works

But if i use

```
global:
  annotations:
      eks.amazonaws.com/role-arn: arn:aws:iam::xxxxxxx:role/xxxxxg-pingelm-pap-irsa
  rbac:
    applyServiceAccountToWorkload: true
    generateGlobalServiceAccount: true
    generateGlobalRoleAndRoleBinding: true
```

Then i get this error

**Error: UPGRADE FAILED: error validating "": error validating data: [apiVersion not set, kind not set]
make: *** [pingdevops] Error 1**

Look like there is some issue with service account templates which are not able to parse annottaions

**Comment by henryrecker-pingidentity** [comment]
It looks like you have a space in your annotation, is that intentional?

**Comment by henryrecker-pingidentity** [comment]
Thanks for opening this issue @mk2134226 , I've opened an issue internally and I'll update this issue when we have a fix.

**Comment by henryrecker-pingidentity** [comment]
A fix for this was merged into main and will be included in the next monthly release

**Comment by mk2134226** [comment]
@henryrecker-pingidentity  Thanks , is there any simple way to add annotation only to service accounts ? rather to all resources ?

**Comment by henryrecker-pingidentity** [comment]
Currently there isn't a way built-in to the chart to do this. You can do this with a kustomize post renderer (similar to https://github.com/pingidentity/pingidentity-devops-getting-started/blob/master/30-helm/read-only-filesystem/kustomize/kustomization.yaml) to set the desired annotation.

**Comment by corybolar** [comment]
@henryrecker-pingidentity This limitation also affects services.  For instance, I need to add a traefik servertransport annotation to the service resources created by the helm chart but can only do so by adding the annotation globally to every resource.  While this doesn't immediately cause an issue, it is very confusing and causes stateful set update issues if the annotation ever changes (The annotation is added to the persistent volume templates).

Is it possible to modify the helm chart to allow annotations for resources that do not currently support sourcing values from the values.yaml?

Unfortunately the kustomize workaround doesn't easily apply as I am using a gitops tool to render the tempaltes, argocd in this case.

**Comment by henryrecker-pingidentity** [comment]
Hi @corybolar , thank you for the feedback. I'll open an internal ticket to track this request and will update you here when it is available.

**Comment by henryrecker-pingidentity** [comment]
We've merged a change to main to allow more fine-grained control over labels and annotations across various resources created by the chart. It'll be included in the next release.

Annotations and labels can now be set per-product for RBAC resources, Workloads, Pod templates in workloads, Services (External and ClusterIP), and HPA. Global values can also be set, such as to set an annotation for all services created by the chart for example.

Here is an example of some of the new annotations that can be set with this change (labels can be set in the same way):
```
global:
  annotations:
    globalAnnotation: val
  
  workload:
    annotations:
      globalWorkloadAnnotation: val
    statefulSet:
      annotations:
        globalSSAnnotation: val
    deployment:
      annotations:
        globalDepAnnotation: val
  
  rbac:
    generateGlobalServiceAccount: true
    generateGlobalRoleAndRoleBinding: true
    serviceAccountAnnotations:
      globalServiceAccountAnnotation: val
    roleAnnotations:
      globalRoleAnnotation: val
    roleBindingAnnotations:
      globalRoleBindingAnnotation: val

  services:
    annotations:
      globalServiceAnnotation: val
    clusterServiceAnnotations:
      globalClusterServiceAnnotation: val

pingdirectory:
  enabled: true
  annotations:
    pdAnnotation: val
  workload:
    annotations:
      pdWorkloadAnnotation: val
    statefulSet:
      annotations:
        pdSSAnnotation: val
  rbac:
    generateServiceAccount: true
    generateRoleAndRoleBinding: true
    role:
      rules:
      - apiGroups: [""]
        resources: ["secrets"]
        verbs: ["get", "list"]
    serviceAccountAnnotations:
      pdServiceAccountAnnotation: val
    roleAnnotations:
      pdRoleAnnotation: val
    roleBindingAnnotations:
      pdRoleBindingAnnotation: val
  services:
    annotations:
      pdServiceAnnotation: val
    clusterServiceAnnotations:
      pdClusterServiceAnnotation: val
```

**Comment by mk2134226** [comment]
@henryrecker-pingidentity  when is new helm chart  getting released , its more than 1 month now , i am waiting for new chnages

**Comment by henryrecker-pingidentity** [comment]
@mk2134226  We've just released version 0.11.0 which includes the new changes - https://helm.pingidentity.com/release-notes/currentRelease/
